src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLogger.cs

```diff
@@ -450,6 +451,15 @@ private void StartTelemetryIfFunctionInvocation(IDictionary<string, object> stat
 
                     // We'll need to store this operation context so we can stop it when the function completes
                     IOperationHolder<RequestTelemetry> operation = _telemetryClient.StartOperation(request);
+
+#pragma warning disable 612, 618
+                    // currently ApplicationInsights supports 2 parallel correlation schemes:
+                    // legacy and W3C, they both appear in telemetry. UX handles all differences in operation Ids. 
+                    // This will be resolved in next .NET SDK on Activity level.
+                    // This ensures W3C context is set on the Activity.
+                    Activity.Current?.GenerateW3CContext();
```

Comment:
Elsewhere we check if options.EnableW3CDistributedTracing is enabled before adding initializers. Do we need to check that here or is this enabling something different?

Reply:
good catch! it's harmless if W3C is disabled, but anyway, let's not do it.